linkml/generators/sqltablegen.py

```diff
@@ -238,73 +239,95 @@ def _sql_name(sn: str):
         schema_metadata.create_all(engine)
         return ddl_str
 
-    def get_sql_range(self, slot: SlotDefinition, schema: SchemaDefinition = None):
+    def get_oracle_sql_range(self, slot: SlotDefinition) -> Text | VARCHAR2 | None:
```

Comment:
Separated out into its own function as the `get_sql_range` function is already pretty complicated.

Reply:
Oof ya, I suspect we will need the notion of "dialect" as p much every SQL abstraction package needs.